react-components/src/architecture/concrete/primitives/tools/PrimitiveEditTool.ts

```diff
@@ -0,0 +1,413 @@
+import { CommandsUpdater } from '../../../base/reactUpdaters/CommandsUpdater';
+import { LineDomainObject } from '../line/LineDomainObject';
+import { CommonRenderStyle } from '../../../base/renderStyles/CommonRenderStyle';
+import { type VisualDomainObject } from '../../../base/domainObjects/VisualDomainObject';
+import { PlaneDomainObject } from '../plane/PlaneDomainObject';
+import { Changes } from '../../../base/domainObjectsHelpers/Changes';
+import { type BaseTool } from '../../../base/commands/BaseTool';
+import { SolidDomainObject } from '../common/SolidDomainObject';
+import { BoxDomainObject } from '../box/BoxDomainObject';
+import { CylinderDomainObject } from '../cylinder/CylinderDomainObject';
+
+export abstract class PrimitiveEditTool extends BaseEditTool {
+  // ==================================================
+  // INSTANCE FIELDS
+  // ==================================================
+
+  private _creator: BaseCreator | undefined = undefined;
+  public primitiveType: PrimitiveType;
+  public readonly defaultPrimitiveType: PrimitiveType;
+
+  public get isEdit(): boolean {
+    return this.primitiveType === PrimitiveType.None;
+  }
+
+  // ==================================================
+  // CONSTRUCTOR
+  // ==================================================
+
+  public constructor(primitiveType: PrimitiveType = PrimitiveType.None) {
+    super();
+    this.primitiveType = primitiveType;
+    this.defaultPrimitiveType = primitiveType;
+  }
+
+  // ==================================================
+  // OVERRIDES of BaseTool
+  // ==================================================
+
+  public override onDeactivate(): void {
+    this.onEscapeKey();
+    super.onDeactivate();
+  }
+
+  public override clearDragging(): void {
+    super.clearDragging();
+    this._creator = undefined;
+  }
+
+  public override onDeleteKey(): void {
+    const domainObject = this.getSelected();
+    if (domainObject !== undefined) {
+      this.addTransaction(domainObject.createTransaction(Changes.deleted));
+      domainObject.removeInteractive();
+    }
+    this._creator = undefined;
+  }
+
+  public override onEscapeKey(): void {
+    if (this._creator !== undefined && this._creator.onEscapeKey()) {
+      this.endCreatorIfFinished(this._creator, true);
+    } else {
+      this.setDefaultPrimitiveType();
+      this._creator = undefined;
+    }
+  }
+
+  public override async onHover(event: PointerEvent): Promise<void> {
+    // Handle when creator is set first
+    if (!this.isEdit) {
+      if (this._creator !== undefined) {
+        const { _creator: creator } = this;
+        if (!creator.preferIntersection) {
+          // Hover in the "air"
+          const ray = this.getRay(event);
+          if (creator.addPoint(ray, undefined, true)) {
+            this.setDefaultCursor();
+            return;
+          }
+        }
+        const intersection = await this.getIntersection(event);
```

Comment:
This code is very hard to follow - would it be easier to make it clear if it was split into functions?

Reply:
Yes it is. I want to want until the annotation is merge. Because I don't  know if everything here is actual wanted